packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -568,6 +574,17 @@ export const Table = <
     ],
   );
 
+  const scrollToIndex = React.useMemo((): number | undefined => {
+    if (!enableVirtualization || paginatorRenderer) {
+      return undefined;
+    }
+    const index = rows.findIndex((row) => row.original == scrollToItem);
```

Comment:
Hmm.. Could you add an example on storybook? Currently I am thinking of passing an item versus passing an index... 

Of course, need to think about filtered out list of items too, so not sure :/

Reply:
Added a simple story https://github.com/iTwin/iTwinUI-react/pull/689/commits/3ef398f1618d1e010c190fdee72ba8add0239186

Reply:
so when I look at your story.. why not make prop accept index instead? We could save all this `findIndex` and custom type too.
I was just not sure the exact flow and if user has index of the item he wants to scroll. Thats why I asked for story as I thought maybe you have an exact use case and would know the situation

Reply:
The consumer isn't necessarily going to know the index. They have the data, but the table may be sorted or filtered. The user generally wants to have a certain _item_ highlighted, not necessarily an _index_.

I have two use cases in my app.
1) A user gets a link to an item. When loading the page, we want the table scrolled so the linked item is visible.
2) A user selects an item, performs an action, and we reload the table with new data. After reloading, we want the table scrolled so the recent item is still visible.

Each of these cases is easy for me, as the items are always selected. So I just pass in `scrollToItem={selectedItems[0]}`. Maybe I could update the story along those lines.

Another consideration is that object equality may not always work if people are doing something tricky with their data. We could query by the the data index by forcing consumers to pass in `getRowId`, and then take the row id instead of the item. I think, however, it works as expect as is.

Reply:
hm.. I think it works fine.
I tried in our playground (just dropped the story code) and removed `getRowId` and I got the error (just needed to explicitly add types to data).
So it might be fine.

Update the tests and lets merge it, if there is nothing else?